src/pages/Course/Course.hooks.tsx

```diff
@@ -0,0 +1,134 @@
+import { Typography, Tooltip, LinearProgress, LinearProgressProps, linearProgressClasses } from '@common/components'
+import { styled } from '@common/theme'
+import { useMemo } from 'react'
+import { StyledComponent } from '@emotion/styled'
+
+export type CourseHookReturn = {
+  readonly calculateTopicProgress: (learningElementProgressTopics: number[][], index: number) => JSX.Element
+  readonly BorderLinearProgress: StyledComponent<
+    LinearProgressProps & { value: number } & { text: string } & {
+      textposition: { xs: string; sm: string; md: string; lg: string; xl: string }
+    }
+  >
+}
+
+/**
+ * # Course Hooks
+ * Hooks for the Course page.
+ * @returns An object with the functions calculateTopicProgress and BorderLinearProgress.
+ * @remarks
+ * Uses the {@link calculatePercent} function to calculate the progress of a topic.
+ * Uses the {@link colorByPercent} function to calculate the color of the progress bar.
+ * Uses the {@link calculateTopicProgress} function to calculate the progress of a topic.
+ * Uses the {@link BorderLinearProgress} styled component to style the progress bar.
+ * @category Hooks
+ */
+export const useCourse = (): CourseHookReturn => {
+  /*
```

Comment:
könntest drüber in der Course Hooks beschreibung als beispiel reinmachen. dann ist es auch in der doku.